state.go

```diff
@@ -0,0 +1,56 @@
+package flagsmith
+
+import (
+	"sync"
+
+	"github.com/Flagsmith/flagsmith-go-client/v4/flagengine/environments"
+	"github.com/Flagsmith/flagsmith-go-client/v4/flagengine/identities"
+)
+
+// environmentState is a locally cached environments.EnvironmentModel.
+type environmentState struct {
+	environment *environments.EnvironmentModel
+	offline     bool
+	mu          sync.RWMutex
+
+	identityOverrides sync.Map
+}
+
+// GetEnvironment returns the current environment and indicates if it was initialised.
+func (cs *environmentState) GetEnvironment() *environments.EnvironmentModel {
+	cs.mu.RLock()
+	defer cs.mu.RUnlock()
+	return cs.environment
+}
+
+func (cs *environmentState) GetIdentityOverride(identifier string) (*identities.IdentityModel, bool) {
```

Comment:
Same as above, we can get rid of the bool here as well?

Reply:
Same reason as above, same fixed :)